cllc-public-app/Controllers/ApplicationsController.cs

```diff
@@ -1097,7 +1097,7 @@ public async Task<IActionResult> UpdateApplication([FromBody] ViewModels.Applica
                     // get entityid
                     string filter = $"_adoxio_application_value eq {id}";
                     int count = _dynamicsClient.Hoursofservices.Get(filter: filter).Value.Count();
-                    MicrosoftDynamicsCRMadoxioHoursofservice hoursEntity = _dynamicsClient.Hoursofservices.Get(filter: filter).Value.First();
+                    MicrosoftDynamicsCRMadoxioHoursofservice hoursEntity = _dynamicsClient.Hoursofservices.Get(filter: filter).Value.FirstOrDefault();
```

Comment:
Can you wrap this in a try - catch - as MS Dynamics may throw an exception rather than return the empty set for a Get that has no results.